### ISSUE TITLE
eth_getStorageAt result is not in hex

### ISSUE THREAD

**Issue by ethers** [ethereum/go-ethereum#517]
On commit 9663493 I had a contract store basically 98+3 in storage[3].

I got "e" and expected "0x65"
```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getStorageAt","params":["0xcdbb08538b8f548b1ad9a1abffd35fb423e91c37", "0x03", "latest"],"id":1}' http://localhost:8080
{
	"id": 1,
	"jsonrpc": "2.0",
	"result": {
		"0x03": "e"
	}
}            
```
looks like Go needs to apply couple of functions to convert?  eg in python >>> hex(ord('e')) gives '0x65'

**Comment by obscuren** [comment]
@tgerring Any idea? Is this old?

**Comment by tgerring** [comment]
Affecting code is likely the block at https://github.com/ethereum/go-ethereum/blob/9663493ba0d493ec62153e725b5d30340f2aa525/rpc/api.go#L312-L322

I see a couple of possible issues with this:
* The call to SetString uses base 10 instead of 16
* value is manipulated to tx, but tx is only used in the logging output
* value is set using `state.StorageString()`, but probably could be set using `common.Bytes2Hex(state.StorageValue())`

@obscuren Any thoughts on the above?

**Comment by tgerring** [comment]
RPC code has been refactored, see https://github.com/ethereum/go-ethereum/blob/16a58f8f2ae860eae28a7d315eb2c9e533c12c3b/rpc/api.go#L110-L122. Can you test on latest commit of develop?

**Comment by obscuren** [comment]
@ethers could you please have a look :)

**Comment by ethers** [comment]
yes, trying... (currently not able to create transactions easily)

**Comment by obscuren** [comment]
Please don't use mist at this time. Stick with the CLI :-)

**Comment by ethers** [comment]
looking better, I think result is missing "0x" prepended to it:
```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getStorageAt","params":["0x11552179f09476f05b85ed7742059e9ba997df45", "0x10" , "latest"],"id":1}' http://localhost:8080
{
	"id": 1,
	"jsonrpc": "2.0",
	"result": "10"
}%                      
```

**Comment by tgerring** [comment]
Fixed in commit 29930da52272b8fd644c38cc303ba6aa66e49182
